lib/ui/login/login_view_model.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_survey/model/login_model.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/ui/login/login_state.dart';
+import 'package:flutter_survey/usecases/log_in_use_case.dart';
+import 'package:email_validator/email_validator.dart';
+
+class LoginViewModel extends StateNotifier<LoginState> {
+  final LogInUseCase _loginUseCase;
+
+  LoginViewModel(
+    this._loginUseCase,
+  ) : super(const LoginState.init());
+
+  void logIn(String email, String password, BuildContext context) async {
+    state = const LoginState.loading();
+    if (_isCredentialsValid(email, password)) {
+      final LoginInput input = LoginInput(
+        email: email,
+        password: password,
+      );
+      final result = await _loginUseCase.call(input);
+      if (result is Success<LoginModel>) {
+        final LoginModel loginData = result.value;
+        _loginUseCase.storeLoginData(loginData);
```

Comment:
Would it be better to handle it directly inside `log_in_use_case.dart` ? 💭 

Then we can minimize the code inside the VM

Reply:
In view model for deciding the state we need to check if the result is a success or failure, if we want to handle `storage task` directly from `usecase` we have to check if the result is a success or failure which will be redundant. I think it is better to take logical decisions from ViewModel, what do you think?

Reply:
> we have to check if the result is a success or failure which will be redundant.

Would you mind explaining this a bit more? 🙇 

I was thinking we could do it like this:

![Screen Shot 2023-04-19 at 11 40 39](https://user-images.githubusercontent.com/18277915/232968848-00b6d9ba-20e2-4569-bf58-3d5608fa0c1e.png)


@luongvo Any thoughts on this? 💭

Reply:
@Tuubz this is good but what do you think about `usecase` performing multiple responsibilities here?

Reply:
Hmmmm, that's also true. 👍 

Then maybe it's best to create a new use-case for storing the login-data? Ex: `StoreTokenUseCase()`

Reply:
@Tuubz let's say we have a new usecase named `StoreTokenUseCase()` should we call it's method to save tokens from `LoginViewModel` or other `LoginUseCase` ?

Reply:
I'm fine with both 🙏